docker-compose.yml

```diff
@@ -0,0 +1,111 @@
+version: '3.5'
+services:
+  app: &app
+    build:
+      context: .
+      target: hyrax-engine-dev
+      args:
+        BUNDLE_WITHOUT: test
+    image: hyrax-engine-dev
+    stdin_open: true
+    tty: true
+    environment:
+      - DATABASE_URL=postgresql://hyrax_user:hyrax_password@postgres/hyrax?pool=5
+      - FCREPO_PORT=8080
+      - FCREPO_HOST=fcrepo
+      - FCREPO_REST_PATH=fcrepo/rest
+      - HYRAX_ENGINE_PATH=/app-data/samvera/hyrax-engine
+      - RACK_ENV=development
+      - RAILS_ENV=development
+      - REDIS_HOST=redis
+      - SOLR_URL=http://solr:8983/solr/hyrax
+    depends_on:
+      - db_migrate
+      - fcrepo
+      - memcached
+      - postgres
+      - redis
+      - solr
+    ports:
+      - 3000:3000
+    volumes:
+      - .dassie:/app-data/samvera/hyrax-webapp
+      - .:/app-data/samvera/hyrax-engine
+
+  db_migrate:
+    image: hyrax-engine-dev
+    environment:
```

Comment:
the duplicate parts between this block and the above can be extracted out to a x-rails block and then included.  The x- means docker-compose will not try to start it and it doesn't have to be complete. https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd

Reply:
i was also considering whether a `.env` wouldn't be a good idea.

since all the current overlap is in environment variables (and the `db_migrate` container should tolerate having the exact same configuration as `app`), do you think that would be better here?

Reply:
I _LOVE_ dotenv and will not argue against it.

Reply:
this is in the latest!